experiment/measurer/measure_worker.py

```diff
@@ -134,37 +170,22 @@ def run_cov_new_units(self):
         self.UNIT_BLACKLIST[self.benchmark] = (
             self.UNIT_BLACKLIST[self.benchmark].union(set(crashing_units)))
 
-    def merge_new_pcs(self) -> List[str]:
-        """Merge new pcs into |self.covered_pcs_filename| and return the list of
-        all covered pcs."""
-
-        # Create the covered pcs file if it doesn't exist yet.
-        if not os.path.exists(self.covered_pcs_filename):
-            filesystem.write(self.covered_pcs_filename, '')
-
-        with open(self.covered_pcs_filename, 'r+') as file_handle:
-            current_pcs = set(
-                pc.strip() for pc in file_handle.readlines() if pc.strip())
-            sancov_files = glob.glob(os.path.join(self.sancov_dir, '*.sancov'))
-            if not sancov_files:
-                self.logger.error('No sancov files.')
-                return list(current_pcs)
-
-            self.logger.info('Sancov files: %s.', str(sancov_files))
-            new_pcs = set(sancov.GetPCs(sancov_files))
-            all_pcs = sorted(list(current_pcs.union(new_pcs)))
-            # Sort so that file doesn't change if PCs are unchanged.
-            file_handle.seek(0)
-            file_handle.write('\n'.join(all_pcs))
+    def merge_new_pcs(self, cycle: int) -> List[str]:
+        """Merge new pcs into and return the list of all covered pcs."""
+        prev_pcs = self.get_prev_covered_pcs(cycle)
+        covered_pcs_state = self.get_covered_pcs_state(cycle)
+        sancov_files = glob.glob(os.path.join(self.sancov_dir, '*.sancov'))
+        if not sancov_files:
+            self.logger.error('No sancov files.')
+            return list(prev_pcs)
```

Comment:
Does this need be sorted ?

Reply:
No. Nothing actually needs sorting, I just do it to make analysis afterwards easier since the files will be the same if the PCs are unchanged if they are sorted.